### PR TITLE
fix: replace the hardcoded default version string (#1836)

### PR DESCRIPTION
**Pull request by annafang-google**
Replace hardcoded default version string with the one specified in version.txt.

### CODE REVIEW

File: proxy/util/cloudsqlutil.go [hunk 1]
```diff
@@ -44,3 +47,25 @@ func SplitName(instance string) (project, region, name string) {
 		return spl[0], spl[1], spl[2]
 	}
 }
+
+// versionString indiciates the version of the proxy currently in use.
+//
+//go:embed version.txt
```
Comment (enocom) [inline proxy/util/cloudsqlutil.go:53]: You'll need to move version.txt to this directory.

Also, we'll need to update the location of this file in `.github/release-please.yml`.

File: proxy/util/cloudsqlutil.go [hunk 2]
```diff
@@ -15,7 +15,10 @@
 // Package util contains utility functions for use throughout the Cloud SQL Auth proxy.
 package util
 
-import "strings"
+import (
+    "strings"
+	_ "embed"
```
Comment (enocom) [inline proxy/util/cloudsqlutil.go:20]: You'll need to run `go fmt ./...` on this code. Let's talk about setting up your favorite editor to do this for you.

**Review by enocom — COMMENTED** [review]
Looks great! A few minor things to fix here.

File: proxy/util/cloudsqlutil.go (hunk 2 above)
Comment (jackwotherspoon) [reply proxy/util/cloudsqlutil.go:20]: @annafang-google if you are using vscode I believe just installing the Go official extension will auto run `go fmt` when you save a file.

Comment (annafang-google) [reply proxy/util/cloudsqlutil.go:20]: Thank you both! Yes I fixed with go fmt command manually this time. Will integrate it with vim 😄 .

File: proxy/util/cloudsqlutil.go (hunk 1 above)
Comment (annafang-google) [reply proxy/util/cloudsqlutil.go:53]: Jacked told me offline that the release-please.yml file in main branch needs to be updated. So created this PR https://github.com/GoogleCloudPlatform/cloud-sql-proxy/pull/1904. Please let me know if that looks OK. Thanks!

File: proxy/util/cloudsqlutil_test.go [hunk 3]
```diff
@@ -14,7 +14,35 @@
...
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestSemanticVersion(t *testing.T) {
+	v, err := os.ReadFile("version.txt")
+	if err != nil {
+		t.Fatalf("failed to read verion.txt: %v", err)
```
Comment (jackwotherspoon) [inline proxy/util/cloudsqlutil_test.go:26]: typo
```suggestion
		t.Fatalf("failed to read version.txt: %v", err)
```

File: proxy/util/cloudsqlutil_test.go [hunk 4]
```diff
@@ -14,7 +14,35 @@
...
+
+	if got := SemanticVersion(); got != want {
+		t.Fatalf("want = %q, got = %q", want, got)
+	}
+}
+
+func TestUserAgentFromVersionString(t *testing.T) {
+	v, err := os.ReadFile("version.txt")
+	if err != nil {
+		t.Fatalf("failed to read verion.txt: %v", err)
```
Comment (jackwotherspoon) [inline proxy/util/cloudsqlutil_test.go:38]: same here
```suggestion
		t.Fatalf("failed to read version.txt: %v", err)
```

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]

File: proxy/util/cloudsqlutil_test.go (hunk 3 above)
Comment (annafang-google) [reply proxy/util/cloudsqlutil_test.go]: Good catch! Updated it. Thanks!

File: proxy/util/cloudsqlutil_test.go (hunk 4 above)
Comment (annafang-google) [reply proxy/util/cloudsqlutil_test.go]: Fixed. Thanks!

**Review by jackwotherspoon — APPROVED** [review]
LGTM ✅  Let's wait and let @enocom take a look on Monday before merging though

File: proxy/util/cloudsqlutil.go [hunk 5]
```diff
@@ -44,3 +47,25 @@ func SplitName(instance string) (project, region, name string) {
 		return spl[0], spl[1], spl[2]
 	}
 }
+
+// versionString indiciates the version of the proxy currently in use.
+//
+//go:embed version.txt
+var versionString string
+
+// metadataString indiciates additional build or distribution metadata.
```
Comment (enocom) [inline proxy/util/cloudsqlutil.go:56]: s/indiciates/indicates/

File: proxy/util/cloudsqlutil.go [hunk 6]
```diff
@@ -44,3 +47,25 @@ func SplitName(instance string) (project, region, name string) {
...
+// semanticVersion returns the version of the proxy in a semver format.
+func SemanticVersion() string {
+	v := strings.TrimSpace(versionString)
+	if metadataString != "" {
+		v += "+" + metadataString
+	}
+	return v
+}
+
+// userAgentFromVersionString returns an appropriate user agent string for identifying this proxy process.
```
Comment (enocom) [inline proxy/util/cloudsqlutil.go:68]: Let's try to wrap at 80 characters here.

**Review by enocom — APPROVED** [review]

File: proxy/util/cloudsqlutil.go (hunk 5 above)
Comment (annafang-google) [reply proxy/util/cloudsqlutil.go]: Good catch! Fixed.

File: proxy/util/cloudsqlutil.go (hunk 6 above)
Comment (annafang-google) [reply proxy/util/cloudsqlutil.go]: Good catch! Fixed.

**Pull request merged**